src/plugins/workspace/public/components/workspace_overview.tsx

```diff
@@ -3,32 +3,53 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import { EuiPageHeader, EuiButton, EuiPanel, EuiSpacer, EuiTitle } from '@elastic/eui';
 import { useObservable } from 'react-use';
 import { of } from 'rxjs';
-
+import { ApplicationStart } from '../../../../core/public';
 import { useOpenSearchDashboards } from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import DeleteWorkspaceModal from './delete_workspace_modal';
 
 export const WorkspaceOverview = () => {
   const {
-    services: { workspaces },
-  } = useOpenSearchDashboards();
+    services: { workspaces, application },
+  } = useOpenSearchDashboards<{ application: ApplicationStart }>();
 
   const currentWorkspace = useObservable(
     workspaces ? workspaces.client.currentWorkspace$ : of(null)
   );
 
+  const workspaceId = currentWorkspace?.id;
+  const workspaceName = currentWorkspace?.name;
+  const [deleteWorkspaceModalVisible, setDeleteWorkspaceModalVisible] = useState(false);
+
+  const deleteWorkspace = () => {
+    if (workspaceId) {
+      workspaces?.client.delete(workspaceId);
```

Comment:
delete should be a async call and we should show a error toast if this API failed.